File: dataset_reduzido/codenet/Python/Code/s906994739.py
n = int(input())
F = [tuple(map(int, input().split())) for _ in range(n)]
P = [tuple(map(int, input().split())) for _ in range(n)]
ans = -10 ** 18

for i in range(1,2**10):
  tmp = 0
  lst = [0]*n
  for j in range(10): #時間帯
        if (i >> j) & 1: #開けるとした時
            for k in range(n):
                lst[k] += F[k][j]
  for k in range(n):
      tmp += P[k][lst[k]]
  ans = max(ans, tmp)
 
 
print (ans)

File: output/Llama-2-100/codenet/Python/Java/s906994739.java
import java.util.Scanner;

public class s906994739 {

public static void main(String[] args) {

int n = Integer.parseInt(sc.nextLine());

List<List<Integer>> F = new ArrayList<>();
List<List<Integer>> P = new ArrayList<>();

for (int i = 0; i < n; i++) {

List<Integer> temp = new ArrayList<>();

for (int j = 0; j < 10; j++) {

if ((i >> j) & 1 == 1) {

for (int k = 0; k < n; k++) {

temp.add(F[k][j]);

}

}

}

F.add(temp);

}

for (int i = 0; i < n; i++) {

List<Integer> temp = new ArrayList<>();

for (int j = 0; j < 10; j++) {

if ((i >> j) & 1 == 1) {

for (int k = 0; k < n; k++) {

temp.add(P[k][F[k][j]]);

}

}

}

P.add(temp);

}

int ans = -1000000000;

for (int i = 1; i < 2 << 10; i++) {

int tmp = 0;

for (int j = 0; j < 10; j++) {

if ((i >> j) & 1 == 1) {

for (int k = 0; k < n; k++) {

tmp += P[k][lst[k]];

}

}

}

ans = Math.max(ans, tmp);

}

System.out.println(ans);

}

}

}
